### PR TITLE
parse http request

### PR DESCRIPTION
**Pull request by oScape**

### CODE REVIEW

**Comment by Stupremee** [comment]
Why don't we use a well-tested crate to parse the http request? That would be much more reliable I think

**Comment by oScape** [comment]
Yes you are right, I will look for it.

File: src/alcazar.rs [hunk 1]
```diff
@@ -1,7 +1,5 @@
-use std::{
-    io::{BufRead, BufReader},
-    net::{IpAddr, Ipv4Addr, SocketAddr, TcpListener},
-};
+use crate::http_request::HttpRequest;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr, TcpListener};
```
Comment (o0Ignition0o) [inline src/alcazar.rs]: There's probably a couple of unused dependencies here, what does clippy say ?

File: src/http_request.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        // If the request is complete we are returning the response in the stream
+        if request_status.is_complete() {
+            match request.path {
+                Some(ref _path) => {
+                    let response = "HTTP/1.1 200 OK\r\n\r\n";
+                    stream.write(response.as_bytes()).unwrap();
+                    stream.flush().unwrap();
+                }
+                None => {
+                    println!("You get bonkerized.");
```
Comment (o0Ignition0o) [inline src/http_request.rs]: Setting up log would be a great followup!

**Review by o0Ignition0o — COMMENTED** [review]
LGTM

File: src/alcazar.rs (hunk 1 above)
Comment (oScape) [reply src/alcazar.rs]: There is no unused dependencies here :)

Comment (oScape) [reply src/alcazar.rs]: Clippy had a problem on `stream.write`, call to replace by `stream.write_all`

**Review by o0Ignition0o — APPROVED** [review]

**Pull request merged**